src/main/java/com/vmware/vipclient/i18n/l2/text/MessageFormat.java

```diff
@@ -272,6 +271,24 @@ private void format(PluralSelectorContext pluralNumber, int i, String argName, M
                 // that formats the number without subtracting the offset.
                 dest.formatAndAppend(pluralNumber.formatter, arg);
             }
+        } else if (argType == ArgType.SIMPLE){
+            String argFormatType = msgPattern.getSubstring(msgPattern.getPart(i++));
+            String argFormatStyle = "";
+            if ((part = msgPattern.getPart(i)).getType() == MessagePattern.Part.Type.ARG_STYLE) {
+                argFormatStyle = msgPattern.getSubstring(part);
+            }
+            int intFormatType  = findKeyword(argFormatType, typeList);
+            switch (intFormatType) {
+                case TYPE_NUMBER:
+                    int intStyle = findKeyword(argFormatStyle, modifierList);
+                    dest.append(new NumberFormatting().format(arg, locale, intStyle));
+                    break;
+                case TYPE_DATE:
+                    dest.append(new DateFormatting().formatDate(arg, PatternProps.trimWhiteSpace(argFormatStyle), locale));// + StringUtil.upperFirstLetter(argFormatType)
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unsupported format type \"" + argFormatType + "\"");
```

Comment:
Yes, but I think here should throw exception instead of log error. Because on one hand it can tell the user the fail reason; on the other hand it make no sense to going on without implementation code.